src/hnsw/build.c

```diff
@@ -531,3 +541,75 @@ void ldb_ambuildunlogged(Relation index)
     // todo::
     elog(ERROR, "hnsw index on unlogged tables is currently not supported");
 }
+
+void ldb_reindex_external_index(Oid indrelid)
+{
+    BlockNumber          HEADER_BLOCK = 0;
+    Relation             index_rel;
+    Buffer               buf;
+    Page                 page;
+    char                *metric_kind;
+    char                *index_name;
+    HnswIndexHeaderPage *headerp;
+    FmgrInfo             reindex_finfo = {0};
+    char    *function_sig = "_lantern_reindex_external_index(text, text, integer, integer, integer, integer)";
```

Comment:
should this function be schema qualified to `lantern`?
Otherwise, a non-root user can just create a function by this name and signature, and have us call it in privileged mode from here